### PR TITLE
fix(weave): Use sdk-local deserializer instead of saved deserializer for known types like Images

### PR DESCRIPTION
**Pull request by andrewtruong**
## Description

https://wandb.atlassian.net/browse/WB-21517

- Prefer using the SDK's local registered deserializer instead of the saved deserializer for known types like Images

## Testing

Unit tests and manual testing

### CODE REVIEW

File: weave/trace/custom_objs.py [hunk 1]
```diff
@@ -51,8 +57,12 @@ def decode_custom_obj(
     encoded_path_contents: Mapping[str, Union[str, bytes]],
     load_instance_op_uri: Optional[str],
 ) -> Any:
-    load_instance_op = None
-    if load_instance_op_uri is not None:
+    if weave_type["type"] in KNOWN_TYPES:
+        serializer = get_serializer_by_id(weave_type["type"])
```
Comment (tssweeney) [inline weave/trace/custom_objs.py:61]: what happens if the serialization changes? It seems like we should first "try" the local version, then fallback to the remote version, then error

Comment (andrewtruong) [reply weave/trace/custom_objs.py:61]: ok yeah that's safer

File: weave/trace/custom_objs.py [hunk 2]
```diff
@@ -52,31 +56,35 @@ def decode_custom_obj(
     encoded_path_contents: Mapping[str, Union[str, bytes]],
     load_instance_op_uri: Optional[str],
 ) -> Any:
-    load_instance_op = None
-    if load_instance_op_uri is not None:
+    _type = weave_type["type"]
+    found_serializer = False
+
+    # First, try to load the object using a known serializer
+    if _type in KNOWN_TYPES:
```
Comment (tssweeney) [inline weave/trace/custom_objs.py:63]: this is still not quite right - we need to try with the local one first, else fallback to the the remote one

Comment (andrewtruong) [reply weave/trace/custom_objs.py:63]: sorry, asked for review too soon before pushing commit -- can you check again

File: weave/trace/custom_objs.py [hunk 3]
```diff
@@ -9,6 +9,19 @@
...
 
 
+class DecodeCustomObjectError(Exception):
+    """An error that occurs while decoding a custom object."""
+
+
+# in future, could generalize as
+# {target_cls.__module__}.{target_cls.__qualname__}
+KNOWN_TYPES = {
+    "Op",
```
Comment (tssweeney) [inline weave/trace/custom_objs.py:19]: does this work for Ops? I thought ops were a special hard coded one? I thought at one time we had special Op handling in `def from_json(obj: Any, project_id: str, server: TraceServerInterface) -> Any:`

Comment (tssweeney) [reply weave/trace/custom_objs.py:19]: I stand corrected. nvm

**Review by tssweeney — APPROVED** [review]

**Pull request merged**